File: ruby/point-mutations/dna.rb
class DNA
  
  def initialize(strand)
    @strand = strand
  end

  def hamming_distance(other_strand)
    (@strand.chars.zip other_strand.chars).count do |nuclecotide, other_nucleotide|
      other_nucleotide && nuclecotide != other_nucleotide
    end
  end

 
end

File: python/point-mutations/dna.py
class DNA(object):
  
    def __init__(self, strand):
        self.strand = strand

    def hamming_distance(self, other_strand):
        return len([(nucleotide, other_nucleotide) 
            for nucleotide, other_nucleotide in self._pairs(other_strand)
            if nucleotide != other_nucleotide
        ])

    def _pairs(self, other_strand):
        return zip(self.strand, other_strand)